File: LeetCode/0003-longest-substring-without-repeating-characters/solution.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int answer = 0;
        set<char> checkString;
        int i = 0;
        int j = 0;
        if(s == ""){
            answer = 0;
        }else{
            while(i < s.size() && j < s.size()){
                if(!checkString.count(s[j])){
                    checkString.insert(s[j]);
                    answer = max(answer,(j-i)+1);
                    j++;
                }else{
                    checkString.erase(s[i]);
                    i++;
                }
            }
        }
        return answer;
    }
};

File: LeetCode/0003-longest-substring-without-repeating-characters/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let answer = 0;
    let letters = {};
    if(s.length == 1) return 1;
    letters[s.charCodeAt(0)] = 1;
    for(let i = 0, j = 1; j<s.length;j++){
        if(letters[s.charCodeAt(j)] == 1){
            while(letters[s.charCodeAt(j)] > 0){
                letters[s.charCodeAt(i)]--;
                i++;
            }
        }
        letters[s.charCodeAt(j)] = 1;
        answer = Math.max(answer,(j-i)+1);
    }
    return answer;
};
